types/time_test.go

```diff
@@ -110,6 +110,9 @@ func TestDateTime(t *testing.T) {
 		{"2020-05-28 23:59:59T T00:00:00", "2020-05-28 23:59:59"},
 		{"2020-10-22 10:31-10:12", "2020-10-22 10:31:10"},
 		{"2018.01.01 01:00:00", "2018-01-01 01:00:00"},
+
+		// For issue 35291
+		{"2020-01-01 12:00:00.123456+05:00", "2020-01-01 07:00:00.123456"},
```

Comment:
why the test can be passed when the line `end=index` not `index-1`  is before？

Reply:
Because  ```unicode.IsPunct(':')``` is ```true```, and ```unicode.IsPunct('+')``` is ```false```.